app/ts/utils/ethereumUtils.ts

```diff
@@ -76,6 +76,17 @@ export function bigintToDecimalStringWithUnknown(value: bigint | undefined, powe
 	return value === undefined ? '?' : bigintToDecimalString(value, power, maxDecimals)
 }
 
+export const isPracticallyInfinite = (value: bigint, power: bigint) => {
+	const googol = 10n ^ 100n
+	return value / (10n ** power) > googol
+}
+
+export function bigintToDecimalStringWithUnknownAndPracticallyInfinite(value: bigint | undefined, power: bigint, maxDecimals?: number): string {
+	if (value === undefined) return '?'
```

Comment:
I feel like a spinner here would be more appropriate, assuming that is the only way to get `undefined` here (feels like it should be).

Reply:
yeah, there's need for spinner in many places. But I haven't gotten you nerd sniped yet to make nice component for spinner management